opcua-types/src/namespaces.rs

```diff
@@ -26,6 +26,23 @@ impl NamespaceMap {
         }
     }
 
+    /// Create a new namespace map from a vec of variant as we get when reading
+    /// the namespace array from the server
+    pub fn new_from_variant_array(array: &[Variant]) -> Self {
```

Comment:
This should probably return an error if any variant isn't a string, that _is_ an error in this case.

Reply:
yes. I think I originally copied that code from somewhere. Can fix now

Reply:
but once again I need to know what kind of error we should return.
If the project does not define errors, we need to make our own or decide to return the error with statuscode that is used several places

Reply:
In this case you can actually define a unit struct for it, or something like `struct UnexpectedVariantType(usize)` containing the index of the variant that failed or something like that. That's what we do some other places.

Reply:
But simple enums are not compatible with stdeeror and create a lot t of issues isn't it?
Then I would rather use thiserror crate which is anyway a dependency

Reply:
I made an MR: As far as I know this is the correct way to do it, but I do not have so much experience with rust